src/components/PostCommentForm.jsx

```diff
@@ -0,0 +1,46 @@
+import { addCommentByArticleId } from "../../api";
+import { useState } from "react";
+import { useContext } from "react";
+import { UserContext } from "../contexts/User";
+
+function PostCommentForm({ id, setComments }) {
+  const [newComment, setNewComment] = useState("");
+  const { user } = useContext(UserContext);
+
+  function handleSubmit(event) {
```

Comment:
Optimistic rendering might be nice to implement here from a UX perspective, (set comments with the new body, username, before making the api request). The way you've approached this is fine too, but either way we definitely need some loading state as it's not currently obvious to a user whether the request is in progress or something has crashed on a slower connection when they click the post button 🧠